docs/proposals/user-cluster-mla.md

```diff
@@ -0,0 +1,380 @@
+strategy used in the Seed it would be one of:
+
+- `NodePort` - would allocate one more NodePort for MLA per user cluster,
+- `Global LoadBalancer` / `One LoadBalancer Per User Cluster` - would allocate one more listener on the LoadBalancer per
+  user cluster + one more NodePort per user cluster,
+- `SNI` (part of
+  the `Tunneling` [expose strategy](https://github.com/kubermatic/kubermatic/blob/master/docs/proposals/http-tunnel-expose-strategy.md)) - would
+  not allocate more LB listeners nor NodePorts. The User Cluster would be connected to a MLA-specific HTTPS
+  endpoint for the given cluster, e.g.,  `https://mla.<<cluster-id>>.<<seed-cluster-name>>.base.domain`.
+
+For this connectivity strategy, all that is needed from an implementation perspective is a
+proper `nodeport-proxy.k8s.io/expose` annotation on the MLA Gateway Deployment resource.
+
+As a potential enhancement, we could consider expanding on
+the [tunneling expose strategy](https://github.com/kubermatic/kubermatic/blob/master/docs/proposals/http-tunnel-expose-strategy.md),
+and tunneling the MLA traffic through the tunneler agents running on the User Cluster worker nodes to reduce the total
+amount of HTTPs connections between the User Cluster and the Seed Cluster (as one Loki Promtail runs on each node in the
+User Cluster and opens its own connection to the Gateway).
+
+### MLA Controller in Seed-Controller-Manager
+
+To automate the configuration of Seed components, we will introduce a new controller into the Seed-Controller-Manager.
+The MLA Controller will manage the following components in the Seed:
+
+- MLA Gateway:
+    - Deploys & configures one MLA Gateway per each User Cluster.
+- Grafana:
+    - Configures a Grafana Organization for each KKP Project.
+    - Adds / removes users to / from Grafana Organizations when users are added / removed to / from KKP Projects.
+    - Configures data sources for individual User Clusters in Grafana Organizations and in the Superuser Organization.
+- IAP:
+    - Deploy and Configures one Ingress and configures IAP per each User Cluster.
+
+### Project and User Information Propagation
+As we mentioned in the previous sections, the MLA controller in the Seed Cluster will configure a Grafana Organization
+for each KKP project, and map KKP users to Grafana Organization users. Currently, the `Project` and `User` objects only
+exist in Master Cluster, and there are two options to propagate Project and User information from Master Cluster to Seed
+Cluster:
+- **Propagate `Project` and `UserProjectBinding` objects from Master Cluster to Seed Cluster.** The advantage of this
+  option
+  is that we can use existing KKP CRDs. The limitation is that those objects will be duplicated in each Seed Cluster.
+- **Create a Propagation CRD.** Instead of propagating `Project` and `UserProjectBinding` objects to Seed Cluster, we
+  can create a CRD that will include Project-User mapping information in Master Cluster:
+
```

Comment:
The SIG API team(@zreigz , @lsviben ) discussed this with @jiachengxu @aborilov and @rastislavs and we concluded that the best option seems to be to propagate `Project` and `UserProjectBinding` from Master to Seed cluster. Because:
- its simpler
- having duplicated `UserProjectBinding` and `Project` on each Seed cluster should not have any effect on KKP

We noted that implementation should take into consideration cases when Master and Seed clusters are on the same cluster.